File: js/수 이어 쓰기 1 .js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = input[0];

const comp = n.length - 1;
let ans = 0;

for (let i=0; i <comp; i++) {
    ans += 9 *(10**i) *(i+1);
}

ans += (parseInt(n) - (10**comp) + 1) * (comp + 1);

console.log(ans);

File: python/수 이어 쓰기 1 .py
# 시간 초과
# import sys

# input = sys.stdin.readline()
# N = int(input)

# ans = 0
# for i in range(1, N+1):
#     ans += len(str(i))

# print(ans)

n = input()

comp = len(n)-1
ans = 0

for i in range(comp):
    ans += 9 * (10**i) * (i+1)

ans += (int(n)-(10**comp)+1) * (comp+1)

print(ans)